ipc/observability/src/traces.rs

```diff
@@ -0,0 +1,103 @@
+// Copyright 2022-2024 Protocol Labs
+// SPDX-License-Identifier: Apache-2.0, MIT
+
+use std::num::NonZeroUsize;
+use tracing::Level;
+pub use tracing_appender::non_blocking;
+pub use tracing_appender::non_blocking::WorkerGuard;
+use tracing_appender::rolling::RollingFileAppender;
+use tracing_subscriber::{fmt, fmt::Subscriber, layer::SubscriberExt, Layer, LogLevel};
+
+use crate::traces_settings::{FileLayerSettings, TracesSettings};
+use crate::tracing_layers::DomainEventFilterLayer;
+
+//
+pub fn create_temporary_subscriber() -> Subscriber {
+    tracing_subscriber::FmtSubscriber::builder()
+        .with_max_level(Level::TRACE)
+        .with_target(false)
+        .with_file(true)
+        .with_line_number(true)
+        .finish()
+}
+
+pub fn set_global_tracing_subscriber(config: &TracesSettings) -> Option<WorkerGuard> {
+    let console_filter = match config.console {
+        Some(console_settings) => console_settings.level_to_filter(),
+        None => LogLevel::default().to_filter(),
+    };
+
+    // log all traces to stderr (reserving stdout for any actual output such as from the CLI commands)
+    let console_layer = fmt::layer()
+        .with_writer(std::io::stderr)
+        .with_target(false)
+        .with_file(true)
+        .with_line_number(true)
+        .with_filter(console_filter);
+
+    let (file_layer, file_guard) = match config.file {
+        Some(file_settings) if file_settings.enabled => {
+            let (non_blocking, file_guard) = non_blocking(create_file_appender(&file_settings));
+
+            let file_layer = fmt::layer()
+                .json()
+                .with_writer(non_blocking)
+                .with_span_events(fmt::format::FmtSpan::CLOSE)
+                .with_target(false)
+                .with_file(true)
+                .with_line_number(true)
```

Comment:
Do we need these in the file appender? I think we can suppress them.

Reply:
I would recommend keeping these logs because all the logs are currently being directed to the file, not just newly emitted events, as we use tracing as a logging library. Therefore, I believe having these logs there is beneficial.